### PR TITLE
Simplify flags for SourceMemberContainerSymbol

### PR DESCRIPTION
**Pull request by agocke**
Recently we ran out of space in the Flags struct inside
SourceMemberContainerSymbol so we created a second 32-bit field to hold
the extra flags. This is not a bad solution, but it means that
DeclarationModifiers, which is in the full field, can't be expanded. I
found it simpler to just split DeclarationModifiers into its own field
since the total size of the struct is the same and it means there's less
complexity in dealing with DeclarationModifiers specifically.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs [hunk 1]
```diff
@@ -29,77 +29,46 @@ private struct Flags
...
-            private int _flags;
-
-            // More flags.
-            //
-            // |                           |zzzz|f|
-            //
-            // f = FlattenedMembersIsSorted.  1 bit.
-            // z = TypeKind. 4 bits.
-            private const int TypeKindOffset = 1;
+            private const int FlattenedMembersIsSortedOffset = FieldDefinitionsNotedOffset + 1;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs]: >1 [](start = 93, length = 1)

Should we use ```FieldDefinitionsNotedSize``` here to follow the pattern?

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs [hunk 2]
```diff
@@ -23,11 +24,11 @@ protected struct Flags
         {
             // We currently pack everything into a 32 bit int with the following layout:
             //
-            // |   |s|r|q|z|xxxxxxxxxxxxxxxxxxxxxxx|wwwww|
+            // | |t|s|r|q|z|xxxxxxxxxxxxxxxxxxxxxxxx|wwwww|
             // 
             // w = method kind.  5 bits.
             //
-            // x = modifiers.  23 bits.
+            // x = modifiers.  24 bits.
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs]: >// x = modifiers.  24 bits. [](start = 12, length = 27)

What is the motivation for the change? It looks like DeclarationModifiers can fit into 23 bits ([https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Declarations/DeclarationModifiers.cs](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Declarations/DeclarationModifiers.cs)).

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs [hunk 3]
```diff
@@ -36,47 +37,53 @@ protected struct Flags
...
 
             private const int MethodKindOffset = 0;
-            private const int DeclarationModifiersOffset = 5;
-
             private const int MethodKindMask = 0x1F;
-            private const int DeclarationModifiersMask = 0x7FFFFF;
+            private const int MethodKindSize = 5;
+
+            private const int DeclarationModifiersOffset = MethodKindSize + MethodKindOffset;
+            private const int DeclarationModifiersSize = 0;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs]: >private const int DeclarationModifiersSize = 0; [](start = 12, length = 47)

This doesn't feel right. The fact that none of the existing tests caught this, makes me uncomfortable about the entire change. We should come up with some kind of test framework to reliably ensure that we are not sharing bits between different "regions".

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs [hunk 4]
```diff
@@ -23,11 +24,11 @@ protected struct Flags
         {
             // We currently pack everything into a 32 bit int with the following layout:
             //
-            // |   |s|r|q|z|xxxxxxxxxxxxxxxxxxxxxxx|wwwww|
+            // | |t|s|r|q|z|xxxxxxxxxxxxxxxxxxxxxxxx|wwwww|
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberMethodSymbol.cs]: > // | |t|s|r|q|z|xxxxxxxxxxxxxxxxxxxxxxxx|wwwww| [](start = 11, length = 48)

It looks like there are more than 32 bits here.

**Comment by agocke** [comment]
@AlekseyTs Lemme mark this for personal review and I'll let you know when I've verified everything's OK.
